auth/cache_test.go

```diff
@@ -0,0 +1,366 @@
+
+	now := time.Now()
+	tok := oauth2.Token{
+		AccessToken:  "TopSecret!",
+		RefreshToken: "SoRefreshing:)",
+		Expiry:       now,
+	}
+
+	assert.NoError(Write(tok))
+
+	cachePointer, err := Read()
+	require.NoError(err)
+
+	expectedCache := Cache{
+		AccessToken:  tok.AccessToken,
+		RefreshToken: tok.RefreshToken,
+		Expiry:       tok.Expiry,
+		MaxAge:       MaxAge,
+	}
+
+	assert.Equal(expectedCache.AccessToken, cachePointer.AccessToken)
+	assert.Equal(expectedCache.RefreshToken, cachePointer.RefreshToken)
+	assert.Equal(expectedCache.Expiry.Format("2006-01-02T15:04:05 -07:00:00"), cachePointer.Expiry.Format("2006-01-02T15:04:05 -07:00:00"))
+	assert.Equal(expectedCache.MaxAge.String(), cachePointer.MaxAge.String())
+	require.NoError(destroy())
+}
+
+func TestGetCredentialPaths_ReturnsPaths(t *testing.T) {
+	os.Setenv(envVarCacheTestMode, "true")
+
+	assert := assertpkg.New(t)
+
+	credentialPath, credentialDir, err := getCredentialPaths()
+	assert.NoError(err)
+
+	userHome, err := os.UserHomeDir()
+	assert.NoError(err)
+	expectedDirectory := filepath.Join(userHome, testDirectory)
+	expectedPath := filepath.Join(userHome, testDirectory, fileName)
+
+	assert.Equal(expectedDirectory, credentialDir)
+	assert.Equal(expectedPath, credentialPath)
+}
+
+func TestJsonToToken_BadJSONThrowsError(t *testing.T) {
+
+	testCases := []struct {
+		name          string
+		rawJSON       []byte
+		expectedError string
+	}{
+		{
+			name:          "empty JSON",
+			rawJSON:       []byte("{}"),
+			expectedError: "failed to get cache access token",
+		},
+		{
+			name:          "invalid JSON",
+			rawJSON:       []byte("Any random string"),
+			expectedError: "failed to unmarshal the raw data to json: invalid character 'A' looking for beginning of value",
+		},
+		{
+			name:          "empty values",
+			rawJSON:       []byte(`{ "access_token": "", "refresh_token": "", "expiry": "", "max_age": "" }`),
+			expectedError: "failed to unmarshal the raw data to json: parsing time \"\\\"\\\"\" as \"\\\"2006-01-02T15:04:05Z07:00\\\"\": cannot parse \"\\\"\" as \"2006\"",
+		},
+		{
+			name:          "empty access token",
+			rawJSON:       []byte(`{ "access_token": "", "refresh_token": "myrefreshtoken", "expiry": "2022-10-20T17:10:59.273429-04:00", "max_age": 86400000000000 }`),
+			expectedError: "failed to get cache access token",
+		},
+		{
+			name:          "empty refresh token",
+			rawJSON:       []byte(`{ "access_token": "myaccesstoken", "refresh_token": "", "expiry": "2022-10-20T17:10:59.273429-04:00", "max_age": 86400000000000 }`),
+			expectedError: "failed to get cache refresh token",
+		},
+		{
+			name:          "emptyexpiry",
+			rawJSON:       []byte(`{ "access_token": "myaccesstoken", "refresh_token": "myrefreshtoken", "expiry": "", "max_age": 86400000000000 }`),
+			expectedError: "failed to unmarshal the raw data to json: parsing time \"\\\"\\\"\" as \"\\\"2006-01-02T15:04:05Z07:00\\\"\": cannot parse \"\\\"\" as \"2006\"",
```

Comment:
I still don't know where this 2006-01-... date is coming from. 🤔

Reply:
https://pkg.go.dev/time?utm_source=godoc#example-Time-Format
It's a constant defined within Golang that helps the compiler understand the format that is being passed in.

Reply:
https://pkg.go.dev/time?utm_source=godoc#pkg-constants